keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -775,129 +788,188 @@ Some flow schemata.
 ```yaml
 kind: FlowSchema
 meta:
-  name: system-top
+  name: exempt
 spec:
-  requestPriority:
-    name: system-top
-  match:
-  - and: # writes by admins (does this cover loopback too?)
-    - superSet:
-      field: groups
-      set: [ "system:masters" ]
+  matchingPrecedence: 1500
+  priorityLevelConfiguration:
+    name: exempt
+  rules:
+  - rule: # All resource requests from system-privileged group
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:masters"
+  - rule: # Delegated auth[n/z] requests
+      verbs: ['*']
+      apiGroups: ['authentication.k8s.io', 'authorization.k8s.io']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:authenticated"     
+    - kind: Group
+      name: "system:unauthenticated"     
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: system-high
+  name: nodes
 spec:
-  requestPriority:
+  matchingPrecedence: 2500
+  priorityLevelConfiguration:
     name: system-high
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # heartbeats by nodes
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: resource
-      value: nodes
-  - and: # kubelet and kube-proxy ops on system objects
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: namespace
-      value: kube-system
-  - and: # leader elections for system controllers
-    - patternMatch:
-      field: user
-      pattern: system:controller:.*
-    - inSet:
-      field: resource
-      set: [ "endpoints", "configmaps", "leases" ]
-    - equals:
-      field: namespace
-      value: kube-system
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule: # All resource requests from node group
```

Comment:
If you look at what the KEP says right now, it calls for "heartbeats by nodes" and "kubelet and kube-proxy ops _on system objects_".  That qualifier "on system objects" was deliberate.  The idea was to give higher priority to maintenance of system things (e.g., pods, services) than for regular things.  But since our current models can not discriminate on object namespace, that qualifier can not be expressed.  That is one of the reasons for https://github.com/kubernetes/kubernetes/pull/81637 .